client/app/routes/index.coffee

```diff
@@ -29,11 +29,9 @@ module.exports = class Router extends Backbone.Router
 
         @listenTo app.model, 'change:dialog', (model, id) ->
             return if id
-            
-            path = if _ref = app.model.get 'filter' then "tags/#{_ref}"
-            else 'contacts'
-            @navigate path, trigger: true
 
+            filter = (app.model.get('filter') or '').match /tag:(\w+)/i
+            @navigate if filter then "tags/#{filter[1]}" else 'contacts'
```

Comment:
> Hard to read, you should probably keep the url / path intermediary variable.

right

> it would be cleaner if the app.model exposed a function with a clear name, like app.model.getCurrentFilteredTag()

Is this really useful? `filter` is just a std backbone.model property, I'm not sure wrapping it in a dedicated getter should be simpler (to maintain nor understand).